insitu/inc/filtered_view.hpp

```diff
@@ -122,6 +122,11 @@ public Q_SLOTS:
 
     void restore(const Json::Value& json);
 
+    // Access selected topic and image dimensions outside insitu core
+    static std::string topic_ref_;
+
+    static std::vector<int> img_dim_;
```

Comment:
In my opinion, `insitu` core should have absolute control over the size of all filters. However, to achieve what you want, you can use the `Set to Image Size` option in the `Filter Properties` pane and read the size in your filter using `getSize()` or `width()` and `height()`. I will consider adding a `Lock to Image Size` boolean that the filter implementors can set to automatically enforce this on filter construction.

Reply:
@PaperFanz This is an okay fix for now. 
However, it defeats the point of what I am trying to do. I do not want the users to have to click on "set to Image size" whenever they change the root image.
Please let us know when you come round to adding a way to automatically enforce this on filter construction.

Reply:
Apologies for the delay in responding. 
I wanted to allow filters to properly scale when the base topic is changed. Eg. filter dimension of 300 * 300 designed on a 1440 * 1440 will look different when put on a 640*480 image. The way I have it (after your suggestion to use getSize() ), the user will have to click on the `Set to Image Size` in order for that filter to update the size. If `Set to image Size`  is set to always be checked by default, this might avoid that. I will push a PR in a few days which will make this clear.